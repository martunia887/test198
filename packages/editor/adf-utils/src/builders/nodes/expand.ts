import { ExpandDefinition, ExpandContent } from '@atlaskit/adf-schema';

export const expand = (attrs: ExpandDefinition['attrs']) => (
  ...content: ExpandContent
): ExpandDefinition => ({
  type: 'expand',
  attrs,
  content,
});
