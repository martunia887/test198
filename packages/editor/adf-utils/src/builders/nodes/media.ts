import {
  MediaDefinition,
  MediaAttributes,
  ExternalMediaAttributes,
} from '@atlaskit/adf-schema';

export const media = (
  attrs: MediaAttributes | ExternalMediaAttributes,
): MediaDefinition => ({
  type: 'media',
  attrs,
});
