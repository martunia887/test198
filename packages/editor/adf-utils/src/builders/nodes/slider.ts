import { SliderDefinition, SliderAttributes } from '@atlaskit/adf-schema';

export const slider = (attrs: SliderAttributes): SliderDefinition => ({
  type: 'slider',
  attrs,
});
