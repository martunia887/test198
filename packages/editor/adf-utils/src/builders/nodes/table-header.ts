import {
  TableHeaderDefinition,
  HeaderCellAttributes,
} from '@atlaskit/adf-schema';

export const tableHeader = (attrs?: HeaderCellAttributes) => (
  ...content: TableHeaderDefinition['content']
): TableHeaderDefinition => ({
  type: 'tableHeader',
  attrs,
  content: content,
});
