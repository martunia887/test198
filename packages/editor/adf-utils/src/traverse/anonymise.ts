import { traverse, ADFNode } from './traverse';

const ANONYMISED_TEXT = 'Content is anonymised';
const ANONYMISED_URL = 'https://anonymised.atlassian.com';
const ANONYMISED_ID = '123456789';
const ANONYMISED_OBJECT = {
  message: 'The content of this object is anonymised',
};

export function anonymise(adf: ADFNode): ADFNode {
  return traverse(adf, {
    any: node => {
      return {
        ...node,
        ...(node.text ? { text: ANONYMISED_TEXT } : {}),
        ...(node.attrs ? { attrs: commonAttributeAnonymiser(node.attrs) } : {}),
      };
    },
  }) as ADFNode;
}

function commonAttributeAnonymiser(attrs: { [name: string]: any }) {
  return {
    ...attrs,
    ...(attrs.id ? { id: ANONYMISED_ID } : {}),
    ...(attrs.text ? { text: ANONYMISED_TEXT } : {}),
    ...(attrs.url ? { url: ANONYMISED_URL } : {}),
    ...(attrs.parameters ? { parameters: ANONYMISED_OBJECT } : {}),
    ...(attrs.shortName ? { shortName: ANONYMISED_TEXT } : {}),
    ...(attrs.collection ? { collection: ANONYMISED_ID } : {}),
    ...(attrs.occurrenceKey ? { occurrenceKey: ANONYMISED_ID } : {}),
  };
}
