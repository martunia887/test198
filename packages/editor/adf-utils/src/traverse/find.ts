import { ADFEntity } from '../types';

export function find(
  adf: ADFEntity,
  callback: (node: ADFEntity) => boolean,
): ADFEntity | undefined {
  return checkNode(adf, callback);
}

function checkNode(
  adfNode: ADFEntity,
  callback: (node: ADFEntity) => boolean,
): ADFEntity | undefined {
  const newNode = { ...adfNode };

  if (callback({ ...newNode })) {
    return newNode;
  }

  if (newNode.content) {
    return newNode.content.find(node => !!checkNode(node, callback));
  }

  return;
}
