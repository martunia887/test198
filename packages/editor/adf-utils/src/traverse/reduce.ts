import { ADFEntity } from '../types';
import { traverse, EntityParent } from './traverse';

export function reduce<T = any>(
  adf: ADFEntity,
  callback: (accunulator: T, node: ADFEntity, parent: EntityParent) => T,
  initial: T,
): T {
  let result = initial;

  traverse(adf, {
    any: (node, parent) => {
      result = callback(result, node, parent);
    },
  });

  return result;
}
