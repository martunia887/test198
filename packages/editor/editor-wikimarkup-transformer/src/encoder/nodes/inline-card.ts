import { NodeEncoder } from '..';
import { Node as PMNode } from 'prosemirror-model';

import { INLINE_CARD_FROM_TEXT_STAMP } from '../../parser/tokenize/issue-key';

import { unknown } from './unknown';

export const inlineCard: NodeEncoder = (node: PMNode): string => {
  if (!node.attrs.url) {
    return unknown(node);
  }

  const match = node.attrs.url.match(INLINE_CARD_FROM_TEXT_STAMP);
  if (!match) {
    return `[${node.attrs.url}|${node.attrs.url}|smart-link]`;
  }

  return `[${match[2]}]`;
};
