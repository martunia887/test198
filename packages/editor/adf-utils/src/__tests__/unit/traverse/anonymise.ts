import { anonymise } from '../../../traverse/anonymise';
import {
  doc,
  blockCard,
  blockQuote,
  p,
  bodiedExtension,
  bulletList,
  listItem,
  codeBlock,
  text,
  date,
  decisionList,
  decisionItem,
  heading,
  emoji,
  extension,
  ul,
  li,
  mediaGroup,
  media,
  mention,
  panel,
} from '../../../builders';
import { inlineCard } from '../../../builders/nodes/inline-card';

describe('Anonymise', () => {
  it('should should anonymise doc node', () => {
    const adf = doc();
    expect(anonymise(adf)).toMatchSnapshot();
  });

  it('should anonymise block-card node', () => {
    const adf = doc(blockCard({ url: 'https://www.atlassian.com' }));
    expect(anonymise(adf)).toMatchSnapshot();
  });

  it('should anonymise blockquote node', () => {
    const adf = doc(blockQuote(p('this is a blockquote')));
    expect(anonymise(adf)).toMatchSnapshot();
  });

  it('should anonymise bodiedExtension node', () => {
    const adf = doc(
      bodiedExtension({
        extensionKey: 'plugin',
        extensionType: 'com.atlassian.fabric',
        layout: 'full-width',
        parameters: {
          text: 'user content',
        },
        text: 'a fabric extension',
      })(),
    );
    expect(anonymise(adf)).toMatchSnapshot();
  });

  it('should anonymise bullet-list node', () => {
    const adf = doc(
      bulletList(listItem([p('a list item ')]), listItem([p('2 list item ')])),
    );
    expect(anonymise(adf)).toMatchSnapshot();
  });

  it('should anonymise code-block node', () => {
    const adf = doc(
      codeBlock({ language: 'javascript' })(text('const i = 0;')),
    );
    expect(anonymise(adf)).toMatchSnapshot();
  });

  it('should anonymise date node', () => {
    const adf = doc(p(date({ timestamp: '1111111111111' })));
    expect(anonymise(adf)).toMatchSnapshot();
  });

  it('should anonymise decision nodes', () => {
    const adf = doc(
      decisionList({ localId: 'xxxxx' })(
        decisionItem({ localId: 'xxx', state: 'DECIDED' })(
          text('this is a decision'),
        ),
      ),
    );
    expect(anonymise(adf)).toMatchSnapshot();
  });

  it('should anonymise emoji node', () => {
    const adf = doc(p(emoji({ id: '111', shortName: 'smile', text: 'smile' })));
    expect(anonymise(adf)).toMatchSnapshot();
  });

  it('should anonymise heading node', () => {
    const adf = doc(heading({ level: 1 })(text('this is heading')));
    expect(anonymise(adf)).toMatchSnapshot();
  });

  it('should anonymise extension node', () => {
    const adf = doc(
      extension({
        extensionKey: 'plugin',
        extensionType: 'com.atlassian.fabric',
        layout: 'full-width',
        parameters: {
          text: 'user content',
        },
        text: 'a fabric extension',
      }),
    );
    expect(anonymise(adf)).toMatchSnapshot();
  });

  it('should anonymise inline-card node', () => {
    const adf = doc(p(inlineCard({ url: 'https://www.atlassian.com' })));
    expect(anonymise(adf)).toMatchSnapshot();
  });

  it('should anonymise list nodes', () => {
    const adf = doc(ul(li([p('this is a list')])));
    expect(anonymise(adf)).toMatchSnapshot();
  });

  it('should anonymise media-group node', () => {
    const adf = doc(
      mediaGroup(
        media({
          type: 'file',
          id: 'xxx',
          collection: 'xxx',
          height: 183,
          width: 183,
          occurrenceKey: 'xxx',
        }),
      ),
    );
    expect(anonymise(adf)).toMatchSnapshot();
  });

  it('should anonymise mention node', () => {
    const adf = doc(p(mention({ id: 'xxx', text: 'tongli' })));
    expect(anonymise(adf)).toMatchSnapshot();
  });

  it('should anonymise panel node', () => {
    const adf = doc(panel({ panelType: 'info' })(p('this is user content')));
    expect(anonymise(adf)).toMatchSnapshot();
  });
});
