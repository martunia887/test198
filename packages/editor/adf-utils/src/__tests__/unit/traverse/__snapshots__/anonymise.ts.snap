// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anonymise should anonymise block-card node 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "url": "https://anonymised.atlassian.com",
      },
      "type": "blockCard",
    },
  ],
  "type": "doc",
  "version": 1,
}
`;

exports[`Anonymise should anonymise blockquote node 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "content": Array [
            Object {
              "text": "Content is anonymised",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "blockquote",
    },
  ],
  "type": "doc",
  "version": 1,
}
`;

exports[`Anonymise should anonymise bodiedExtension node 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "extensionKey": "plugin",
        "extensionType": "com.atlassian.fabric",
        "layout": "full-width",
        "parameters": Object {
          "message": "The content of this object is anonymised",
        },
        "text": "Content is anonymised",
      },
      "content": Array [],
      "type": "bodiedExtension",
    },
  ],
  "type": "doc",
  "version": 1,
}
`;

exports[`Anonymise should anonymise bullet-list node 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "Content is anonymised",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "Content is anonymised",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
      ],
      "type": "bulletList",
    },
  ],
  "type": "doc",
  "version": 1,
}
`;

exports[`Anonymise should anonymise code-block node 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "language": "javascript",
      },
      "content": Array [
        Object {
          "text": "Content is anonymised",
          "type": "text",
        },
      ],
      "type": "codeBlock",
    },
  ],
  "type": "doc",
  "version": 1,
}
`;

exports[`Anonymise should anonymise date node 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "attrs": Object {
            "timestamp": "1111111111111",
          },
          "type": "date",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
  "version": 1,
}
`;

exports[`Anonymise should anonymise decision nodes 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "localId": "xxxxx",
      },
      "content": Array [
        Object {
          "attrs": Object {
            "localId": "xxx",
            "state": "DECIDED",
          },
          "content": Array [
            Object {
              "text": "Content is anonymised",
              "type": "text",
            },
          ],
          "type": "decisionItem",
        },
      ],
      "type": "decisionList",
    },
  ],
  "type": "doc",
  "version": 1,
}
`;

exports[`Anonymise should anonymise emoji node 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "attrs": Object {
            "id": "123456789",
            "shortName": "Content is anonymised",
            "text": "Content is anonymised",
          },
          "type": "emoji",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
  "version": 1,
}
`;

exports[`Anonymise should anonymise extension node 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "extensionKey": "plugin",
        "extensionType": "com.atlassian.fabric",
        "layout": "full-width",
        "parameters": Object {
          "message": "The content of this object is anonymised",
        },
        "text": "Content is anonymised",
      },
      "type": "extension",
    },
  ],
  "type": "doc",
  "version": 1,
}
`;

exports[`Anonymise should anonymise heading node 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "level": 1,
      },
      "content": Array [
        Object {
          "text": "Content is anonymised",
          "type": "text",
        },
      ],
      "type": "heading",
    },
  ],
  "type": "doc",
  "version": 1,
}
`;

exports[`Anonymise should anonymise inline-card node 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "attrs": Object {
            "url": "https://anonymised.atlassian.com",
          },
          "type": "inlineCard",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
  "version": 1,
}
`;

exports[`Anonymise should anonymise list nodes 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "Content is anonymised",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
      ],
      "type": "bulletList",
    },
  ],
  "type": "doc",
  "version": 1,
}
`;

exports[`Anonymise should anonymise media-group node 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "attrs": Object {
            "collection": "123456789",
            "height": 183,
            "id": "123456789",
            "occurrenceKey": "123456789",
            "type": "file",
            "width": 183,
          },
          "type": "media",
        },
      ],
      "type": "mediaGroup",
    },
  ],
  "type": "doc",
  "version": 1,
}
`;

exports[`Anonymise should anonymise mention node 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "attrs": Object {
            "accessLevel": "",
            "id": "123456789",
            "text": "Content is anonymised",
          },
          "type": "mention",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
  "version": 1,
}
`;

exports[`Anonymise should anonymise panel node 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "panelType": "info",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "text": "Content is anonymised",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "panel",
    },
  ],
  "type": "doc",
  "version": 1,
}
`;

exports[`Anonymise should should anonymise doc node 1`] = `
Object {
  "content": Array [],
  "type": "doc",
  "version": 1,
}
`;
